pkg/detector/policy_test.go

```diff
@@ -37,6 +37,829 @@ import (
+					Object: map[string]interface{}{
+						"apiVersion": "apps/v1",
+						"kind":       "Deployment",
+						"metadata": map[string]interface{}{
+							"name":      "deployment",
+							"namespace": "test",
+						},
+					},
+				}
+				rb := &workv1alpha2.ResourceBinding{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:            "binding-1",
+						ResourceVersion: "999",
+						Labels: map[string]string{
+							policyv1alpha1.PropagationPolicyPermanentIDLabel: "f2507cgb-f3f3-4a4b-b289-5691a4fef979",
+						},
+						Annotations: map[string]string{
+							policyv1alpha1.PropagationPolicyNamespaceAnnotation: "deploy-match-namespace-1",
+							policyv1alpha1.PropagationPolicyNameAnnotation:      "deploy-match-name-1",
+						},
+					},
+					Spec: workv1alpha2.ResourceBindingSpec{
+						Resource: workv1alpha2.ObjectReference{
+							APIVersion: "apps/v1",
+							Kind:       "Deployment",
+							Namespace:  "test",
+							Name:       "deployment",
+						},
+					},
+				}
+				return fake.NewClientBuilder().WithScheme(scheme).WithObjects(obj, rb).WithRESTMapper(restMapper)
+			},
+			existingObject: &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"apiVersion": "apps/v1",
+					"kind":       "Deployment",
+					"metadata": map[string]interface{}{
+						"name":      "deployment",
+						"namespace": "test",
+					},
+				},
+			},
+		},
+		{
+			name:            "cannot list unmatched binding resource with policy and namespace",
+			policyID:        "f2507cgb-f3f3-4a4b-b289-5691a4fef979",
+			policyName:      "test-policy-1",
+			policyNamespace: "fake-namespace-2",
+			selectors: []policyv1alpha1.ResourceSelector{
+				{
+					APIVersion: "apps/v1",
+					Kind:       "Pod",
+					Namespace:  "default",
+				},
+			},
+			wantErr: true,
+			setupClient: func() *fake.ClientBuilder {
+				return fake.NewClientBuilder().WithRESTMapper(restMapper)
+			},
+			existingObject: &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"apiVersion": "apps/v1",
+					"kind":       "Deployment",
+					"metadata": map[string]interface{}{
+						"name":      "deployment",
+						"namespace": "test",
+					},
+				},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			fakeClient := tt.setupClient().Build()
+			stopCh := make(chan struct{})
+			defer close(stopCh)
+			fakeDynamicClient := dynamicfake.NewSimpleDynamicClient(scheme, tt.existingObject)
+			genMgr := genericmanager.NewSingleClusterInformerManager(fakeDynamicClient, 0, stopCh)
+			resourceDetector := &ResourceDetector{
+				Client:          fakeClient,
+				DynamicClient:   fakeDynamicClient,
+				RESTMapper:      fakeClient.RESTMapper(),
+				InformerManager: genMgr,
+			}
+			err := resourceDetector.cleanPPUnmatchedRBs(tt.policyID, tt.policyNamespace, tt.policyName, tt.selectors)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("cleanPPUnmatchedRBs() error = %v, wantErr %v", err, tt.wantErr)
+			}
```

Comment:
In addition to determining whether err is returned, can we add more checks to determine whether the target object has been cleaned up?

Reply:
Sure, let me check...